core/kernel.rbs

```diff
@@ -2447,7 +2445,7 @@ module Kernel : BasicObject
   #     b.instance_of? B   #=> true
   #     b.instance_of? C   #=> false
   #
-  def instance_of?: (Module) -> bool
+  def instance_of?: (Module | Class module_or_class) -> bool
```

Comment:
Technically `Class < Module`, and the `| Class` is redundant.

Reply:
See https://github.com/ruby/rbs/pull/1943/files#r1699454965

Reply:
While the `| Class` is technically redundant, i think it works well as a documentation—new ruby users might not know that all classes are modules, and might be confused as to why `is_a?` and friends don't say they accept `Class`es

Reply:
Has there been discussion in Ruby Core on making `Class` not outright inherit from `Module`?

`Class` is not a strict subtype of `Module` because of the many places that can complain “wrong argument type Class (expected Module) (TypeError)”.
```ruby
include Class
```